includes/data-port/class-sensei-data-port-manager.php

```diff
@@ -123,11 +123,28 @@ public function run_data_port_job( $args ) {
 
 		if ( null !== $job ) {
 			wp_set_current_user( $job->get_user_id() );
-			Sensei_Scheduler::instance()->run( $job );
+			$this->run_data_port_job( $job );
 			wp_set_current_user( 0 );
 		}
 	}
 
+	/**
+	 * Runs a data port job if it's not currently running.
+	 *
+	 * @param Sensei_Data_Port_Job $job
+	 */
+	public function run_data_port_job( $job ) {
+		$job_running = get_option( self::OPTION_RUNNING_JOB, false );
+		if ( ! $job_running ) {
```

Comment:
I added a check to prevent the polling request and scheduler both starting a job parallel, as they would then process the same lines.

Reply:
Good idea. I had an issue with transients not working properly at first, but can't reproduce it now. Updated in 2886346, with a fixed 120s timeout, as the data-port jobs have hard coded batch sizes for now and should be done in 1-5 seconds.

Reply:
That works for me. There is one caveat in that transients are never guarenteed (they are pretty pluggable in WP), but so far for similar implementations in WPJM / Sensei I haven't heard of any reports.